File: Lesson/Lesson 15 Photosensitive resistance component experiment/Java/Photoresistance.java
import com.pi4j.wiringpi.I2C;
import com.pi4j.wiringpi.Gpio;

public class Photoresistance {
    static int led_pin = 2, value = 0, fd;

    static {
        Gpio.wiringPiSetup();
        fd = I2C.wiringPiI2CSetup(0x04);
        Gpio.pinMode(led_pin, Gpio.OUTPUT);
    }

    public static void main(String[] args){
        for ( ; ;){
            value =  I2C.wiringPiI2CReadReg16(fd, 0x10);
            if(value > 1700) {
                Gpio.digitalWrite(led_pin, Gpio.HIGH);
                Gpio.delay(200);
            } else {
                Gpio.digitalWrite(led_pin, Gpio.LOW);
            }
        }

    }
}

File: Lesson/Lesson 15 Photosensitive resistance component experiment/Python/Photoresistance.py
import time
import smbus as smbus
import RPi.GPIO as GPIO

ledpin = 27
ADC=smbus.SMBus(1)#Declare to use I2C 1
GPIO.setmode(GPIO.BCM)  
GPIO.setup(ledpin, GPIO.OUT)
    
while True:
    ADC.write_byte(0x04,0x20)#Write a byte to the slave
    val = ADC.read_word_data(0x04,0x20);
    temp = val / 1023 * 5 * 100  
    if temp>0:
        GPIO.output(ledpin, True)
        time.sleep(1)
    else:
        GPIO.output(ledpin, False)
    print(temp)#Raspberry Pi reads the data returned by the expansion board and prints it out  
    time.sleep(1)#Delay 1 second
